feat: Linux Flatpak Support (x86_64)

GillesHorn opened the pull request:
## Description
This PR introduces full Flatpak support for Sharly Chess (x86_64).

## Changes
- **CI/CD**: Added \.github/workflows/linux-flatpak.yml\ for automated building and testing.
- **Manifest**: Created \latpak/configuration/com.sharlychess.SharlyChess.json\.
- **Scripts**: Added build, test, and initialization scripts in \latpak/scripts/\ and \scripts/\.
- **Documentation**: Comprehensive documentation in \latpak/documentation/\ including:
    - User Guide
    - Developer Guide
    - CI/CD Setup
    - Migration Guide
- **Secrets**: Implemented secure secret management for Chess-Results and FFE integration.
- **Python**: Upgraded to Python 3.13.

## Verification
- [x] CI/CD Pipeline passes.
- [x] Documentation is complete and up-to-date.
- [x] Secrets are handled securely.

Amaras commented on .github/workflows/linux-flatpak.yml:
```diff
@@ -0,0 +1,149 @@
+on:
+  push:
+    tags:
+      - 'v*'
```
Not our tag format

GillesHorn replied:
on:
  push:
    branches: [ dev, 'hotfix/*', 'release/*' ]
  pull_request:
    branches: [ dev, 'hotfix/*', 'release/*' ]

Amaras commented on .github/workflows/publish-multiarch.yml:
```diff
@@ -0,0 +1,336 @@
+    branches:
+      - linux-flatpak
+    tags:
+      - 'v*'
```
not our tag format

GillesHorn replied:
on:
  push:
    branches:
      - linux-flatpak
    tags:
      - '*'

Amaras commented on flatpak/configuration/INSTALL.md:
```diff
@@ -0,0 +1,86 @@
+Download the `com.sharlychess.SharlyChess.flatpak` file from the GitHub Actions artifacts or your build output.
+
+### 2. Install Dependencies (Runtimes)
+Sharly Chess requires the GNOME Platform version 48. While Flatpak usually handles dependencies automatically, installing them explicitly can prevent issues.
```
Are we on version 48, or 49?

GillesHorn replied:
Changed in the documentation to Gnome 49

Amaras commented on .github/workflows/build-test-with-init.yml:
```diff
@@ -0,0 +1,86 @@
+          
+      - name: Build application
+        run: |
+          python -m PyInstaller --onefile src/sharly_chess.py
```
we use `--onedir`, not `--onefile` due to startup performance issues

GillesHorn replied:
modification done for --onedir

Amaras commented on .github/workflows/build-test-with-init.yml:
```diff
@@ -0,0 +1,86 @@
+  push:
+    branches: [ dev ]
+  pull_request:
+    branches: [ dev ]
```
You don't want to test the build on `hotfix/*` and `release/*` branches?

GillesHorn replied:
removed build-test-with-ini.yml, added [branch ](branches: [ dev, 'hotfix/*', 'release/*' ]) to lniux-flatpak*

Amaras commented on .github/workflows/linux-flatpak-arm64.yml:
```diff
@@ -0,0 +1,150 @@
+on:
+  push:
+    tags:
+      - 'v*'
```
That's not our tag format

GillesHorn replied:
modified by 
on:
  push:
    branches: [ dev, 'hotfix/*', 'release/*' ]
  pull_request:
    branches: [ dev, 'hotfix/*', 'release/*' ]
  workflow_dispatch:

Amaras reviewed (changes requested):
Okay, first pass with general build remarks

Commit:
Feat: Linux Flatpak Support (x86_64 & ARM64)

This commit introduces full Flatpak support for Sharly Chess.

Key Features:
- Flatpak Manifests: Configuration for building the app in a sandboxed environment.
- CI/CD Workflows: GitHub Actions for building, testing, and publishing (x86_64 & ARM64).
- Python 3.13 Support: Updates to codebase and dependencies for compatibility.
- Documentation: Comprehensive guides in flatpak/documentation/.
- Scripts: Launchers, migration tools, and validation scripts.

Technical Details:
- Uses GNOME 49 Runtime.
- Implements 'flatpak build-commit-from' for OSTree history preservation.
- Secure handling of secrets for FFE/Chess-Results integration.
- Automatic data migration from legacy versions.
